### DISCUSSION TITLE
multi-file autocompletion, intellisense, goto-def

### DISCUSSION THREAD

**Discussion by ide5kllc** [microsoft/monaco-editor#3718] [Q&A]
### Context

- [X] This issue is not a bug report. *(please use a different template for reporting a bug)*
- [X] This issue is not a duplicate of an existing issue. *(please use the [search](https://github.com/microsoft/monaco-editor/issues) to find existing issues)*

### Description

I am trying to figure out how to do multi-file autocompletion, intellisense, goto-def.

I have done the following:

1. created a model for every file, via monaco.editor.createModel
2. verified that the above models work (clicking on div changes file/model) via active_editor.setModel(...)

However, I do not have auto complextion/goto-def working across files.

Are there (up-to-date) instructions anywhere on how to make this work?

**Comment by pennywort** [comment]
Try to add them as libs. However it should work without adding libs. May be you need to pass correct uri in createModel().
Try this code: 
```typescript
monaco.languages.typescript.javascriptDefaults.setDiagnosticsOptions({
    noSemanticValidation: true,
    noSyntaxValidation: false
});
 
monaco.languages.typescript.javascriptDefaults.setCompilerOptions({
    target: monaco.languages.typescript.ScriptTarget.ES6,
    allowNonTsExtensions: true
});
 
// extra libraries
var libSource =
`
interface Application {
    /**
     * Test JSDOC
     */
  onLaunch(): void;
}
 
interface ApplicationFactory {
  extends(): Application;
  hui: String;
}
 
interface ControllerFactory {
  extends(): Controller;
}
 
interface Controller {
    /**
     * asdf afads fasd fadsf asdf adsfads f
     */
  onLoad(): void;
}
 
type Registry = {
  "ac/1": ApplicationFactory;
  "ac/2": ControllerFactory;
};
 

const define = (depNames: ["ac/1", "ac/2"], factory: (a: ApplicationFactory, b: ControllerFactory) => ) => void;
`

var libUri = 'ts:filename/facts.d.ts';
monaco.languages.typescript.javascriptDefaults.addExtraLib(libSource, libUri);
monaco.editor.createModel(libSource, 'typescript', monaco.Uri.parse(libUri));
 
var jsCode = `define(['ac/1', 'ac/2'], (a, b) => {
    //fill me
    a.
    a.extends().onLaunch
})
//hover over me
define()`
 
monaco.editor.create(document.getElementById('container'), {
    value: jsCode,
    language: 'javascript'
});
 
```
